provider/oauth2/tests.py

```diff
@@ -597,6 +597,25 @@ def test_access_token_response_valid_token_type(self):
         token = self._login_authorize_get_token()
         self.assertEqual(token['token_type'], constants.TOKEN_TYPE, token)
 
+    def test_client_credentials_grant(self):
+        response = self.client.post(self.access_token_url(), {
+            'grant_type': 'client_credentials',
+            'client_id': self.get_client().client_id,
+            'client_secret': self.get_client().client_secret,
+        })
+
+        self.assertEqual(200, response.status_code, response.content)
```

Comment:
Ah. This is why I wanted someone intimately familiar with the spec reviewing. This code is being adapted from caffeinehit/django-oauth2-provider#47. I admin I haven't fully read the spec, so I was unaware of the need to omit the refresh token. That is great, because testing that has proved frustrating!